test/connectivity/modules/NetworkController/Settings.cs

```diff
@@ -9,24 +9,25 @@ namespace NetworkController
 
     class Settings
     {
+        internal static Settings Current = Create();
         const string StartAfterPropertyName = "StartAfter";
         const string DockerUriPropertyName = "DockerUri";
         const string FrequencyPropertyName = "RunFrequencies";
         const string NetworkIdPropertyName = "NetworkId";
-        const string NetworkControllerModePropertyName = "NetworkControllerMode";
         const string TestResultCoordinatorEndpointPropertyName = "TestResultCoordinatorEndpoint";
         const string TrackingIdPropertyName = "TrackingId";
         const string ModuleIdPropertyName = "IOTEDGE_MODULEID";
         const string IotHubHostnamePropertyName = "IOTEDGE_IOTHUBHOSTNAME";
-
-        internal static Settings Current = Create();
+        const string NetworkControllerRunProfilePropertyName = "NetworkControllerRunProfile";
+        const string DefaultProfilesPropertyName = "DefaultProfiles";
 
         Settings(
             TimeSpan startAfter,
             string dockerUri,
             string networkId,
             IList<Frequency> frequencies,
-            NetworkControllerMode mode,
+            string networkRunProfile,
+            IDictionary<string, NetworkProfile> profileSettings,
```

Comment:
why do we need to pass a dictionary here?

Reply:
To map the profile setting name with the NetworkControllerRunProfile